src/stream.rs

```diff
@@ -416,6 +416,7 @@ pub(crate) fn connect_host(unit: &Unit, hostname: &str, port: u16) -> Result<Tcp
             None => None,
         };
 
+        println!("connecting to {}", &sock_addr);
```

Comment:
This looks like debugging - maybe use `log::debug!` instead? (or just remove it altogether if you don't want to introduce a log dependency.)

Reply:
We actually just added `log` as a dependency!